sys/linux/fs_ioctl_btrfs.txt

```diff
@@ -1,7 +1,443 @@
+}
+
+btrfs_ioctl_received_subvol_args {
+	uuid		array[int8, BTRFS_UUID_SIZE]
+	stransid	transid
+	rtransid	transid
+	stime		btrfs_ioctl_timespec
+	rtime		btrfs_ioctl_timespec
+	flags		int64
+	reserved	array[int64, 16]
+}
+
+btrfs_ioctl_timespec {
+	sec	int64
+	msec	int32
+}
+
+##commented out because I don't know what is the best way to include their corresponding Macros
+# btrfs_ioctl_received_subvol_args_32 {
+# 	uuid 		array[int8, BTRFS_UUID_SIZE]
+# 	stransid 	transid
+# 	rtransid 	transid
+# 	stime 		btrfs_ioctl_timespec_32
+# 	rtime  		btrfs_ioctl_timespec_32
+# 	flags 		int64
+# 	reserved 	array[int64, 16]
+# }
+
+# btrfs_ioctl_timespec_32 {
+# 	sec 	int64
+# 	msec 	int32
+# } [packed]
+
+btrfs_ioctl_send_args {
+	send_fd			align64[fd]
+	clone_sources_count	len[clone_sources, int64]
+	clone_sources		ptr[in, array[subvolid]]
+	parent_root		int64
+	flags			flags[btrfs_ioctl_send_flags, int64]
+	reserved		array[int64, 4]
+}
+
+#This pads the struct to 1032 bytes. It was originally meant to pad to
+#1024 bytes, but when adding the flags field, the padding calculation
+#was not adjusted.
+define BTRFS_DEV_PAD	128-2-BTRFS_DEV_STAT_VALUES_MAX
+
+btrfs_ioctl_get_dev_stats {
+	devid		devid
+	nr_items	int64
+	flags		flags[btrfs_ioctl_get_dev_stats_flags, int64]
+	values		array[int64, BTRFS_DEV_STAT_VALUES_MAX]
+	unused		array[int64, BTRFS_DEV_PAD]
+}
+
+btrfs_ioctl_quota_ctl_args {
+	cmd	flags[btrfs_ioctl_quota_ctl_cmd, int64]
+	status	int64
+}
+
+btrfs_ioctl_qgroup_assign_args {
+	assign	bool64
+	src	qgroupid
+	dst	qgroupid
+}
+
+btrfs_ioctl_qgroup_create_args {
+	create		bool64
+	qgroupid	qgroupid
+}
+
+btrfs_ioctl_qgroup_limit_args {
+	qgroupid	qgroupid
+	lim		btrfs_qgroup_limit
+}
+
+btrfs_ioctl_quota_rescan_args {
+	flags		const[0, int64]
+	progress	int64
+	reserved	array[int64, 6]
+}
+
+btrfs_ioctl_dev_replace_args_u [
+	start	btrfs_ioctl_dev_replace_start_params
+	status	btrfs_ioctl_dev_replace_status_params
+]
+
+btrfs_ioctl_dev_replace_args {
+	cmd	flags[btrfs_ioctl_dev_replace_args_cmd, int64]
+	result	int64
+	union1	btrfs_ioctl_dev_replace_args_u
+	spare	array[int64, 64]
+}
+
+define BTRFS_DEVICE_MAX	BTRFS_DEVICE_PATH_NAME_MAX + 1
+
+btrfs_ioctl_dev_replace_start_params {
+	srcdevid			devid
+	cont_reading_from_srcdev_mode	flags[cont_reading_from_srcdev_mode_flags, int64]
+	srcdev_name			array[int8, BTRFS_DEVICE_MAX]
+	tgtdev_name			array[int8, BTRFS_DEVICE_MAX]
+}
+
+#this structure is just for out, hence no need to describe
+btrfs_ioctl_dev_replace_status_params {
+	dummys	array[int64, 6]
+}
+
+btrfs_ioctl_feature_flags {
+	compat_flags	flags[btrfs_ioctl_feature_compat_flags, int64]
+	compat_ro_flags	flags[btrfs_ioctl_feature_compat_ro_flags, int64]
+	incompat_flags	flags[btrfs_ioctl_feature_incompat_flags, int64]
+}
+
+define BTRFS_VOL_MAX	BTRFS_VOL_NAME_MAX+1
+
+#described since the uuid, transid and treeid is going to be useful as resource
+btrfs_ioctl_get_subvol_info_args {
+	treeid		treeid
+	name		array[int8, BTRFS_VOL_MAX]
```

Comment:
We can probably do better than `array[int8]` here and for other names in other structs.  But maybe it would be easier to review that in a later PR.

Reply:
Yep I'll put them in later PR.